### PR TITLE
Add new classes to models.ts and create a login function

### PR DESCRIPTION
**Pull request by dominoweir**

### CODE REVIEW

File: src/evergreen.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ export class client {
      * @returns nothing
      */
     public getResource(callback: request.RequestCallback, resource: string, params?: object) {
-        const url = this.serverURL + "/" + resource + queryString(params);
+        const url = resource + queryString(params);
```
Comment (john-m-liu) [inline src/evergreen.ts]: What was the reason for changing this? The base server URL needs to be there to send the request

File: tsconfig.json [hunk 2]
```diff
@@ -12,7 +12,7 @@
       "forceConsistentCasingInFileNames": true,
       "noImplicitReturns": true,
       "noImplicitThis": true,
-      "noImplicitAny": true,
+      "noImplicitAny": false,
```
Comment (john-m-liu) [inline tsconfig.json]: We should leave this check in unless there's a good reason to do so

**Comment by john-m-liu** [comment]
I configured pull request testing after you opened this and pushed an empty commit to trigger them to run. Could you take a look at the failures? The test failure is likely a result of not using the server URL

File: src/evergreen.ts (hunk 1 above)
Comment (dominoweir) [reply src/evergreen.ts]: We changed this for testing purposes when I was checking to see if the models I built for the patches actually worked. I will change it back and try running the tests again.

File: tsconfig.json (hunk 2 above)
Comment (dominoweir) [reply tsconfig.json]: Fixed

**Pull request merged**